.github/workflows/back-to-development.yaml

```diff
@@ -0,0 +1,54 @@
+name: Go back to Development
+on:
+  workflow_dispatch:
+  release:
+    types: [published]
+
+jobs:
+  post-release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Set RELEASED_VERSION_WITH_V variable from bufcli.go
+        # in the echo command, globbing patterns are used in the parameter substitution  not regex patterns
+        run: |
+          RELEASED_VERSION_WITH_V=`grep -oE "Version.*=.*\"[0-9]\.[0-9]+\.[0-9]+[^\"]*" private/buf/bufcli/bufcli.go`
+          echo "RELEASED_VERSION_WITH_V=v${RELEASED_VERSION_WITH_V##Version*=*\"}" >> $GITHUB_ENV
+      - name: Set NEXT VERSION variable from tag
+        run: |
+          NEXT_VERSION=$(echo ${{ env.RELEASED_VERSION_WITH_V }} | awk -F. -v OFS=. '{$NF += 1 ; print}')
+          echo "NEXT_VERSION=${NEXT_VERSION:1}-dev" >> $GITHUB_ENV
+      - name: Update version
+        run: make updateversion VERSION=${{ env.NEXT_VERSION }}
+      - name: Unrelease changelog
+        run: |
+          sed -i "/^# Changelog/ {
+          N;
+          a\
+          ## [Unreleased]\\
+          \\
+          - No changes yet.\\
+          
+          }" CHANGELOG.md
+          sed -i "/^Initial beta release.$/ {
+          N;
+          a\
+          [Unreleased]: https://github.com/bufbuild/buf/compare/${{ env.RELEASED_VERSION_WITH_V }}...HEAD
+          }" CHANGELOG.md
+      - name: update home brew badge
+        run: make updatehomebrewbadge VERSION=${{ env.RELEASED_VERSION_WITH_V }}
```

Comment:
I didn't test this in my own repo, but it runs in the same relative order as before. @doriable does this look correct?